lib/logstash/filters/elasticsearch.rb

```diff
@@ -176,6 +176,19 @@ def filter(event)
     end
   end # def filter
 
+  # public only to be reuse in testing
+  def prepare_user_agent
+    os_name = java.lang.System.getProperty('os.name')
+    os_version = java.lang.System.getProperty('os.version')
+    os_arch = java.lang.System.getProperty('os.arch')
+    jvm_vendor = java.lang.System.getProperty('java.vendor')
+    jvm_version = java.lang.System.getProperty('java.version')
+
+    plugin_version = Gem.loaded_specs['logstash-filter-elasticsearch'].version
+    # example: Logstash/7.14.1 (OS=Linux-5.4.0-84-generic-amd64; JVM=AdoptOpenJDK-11.0.11) logstash-output-elasticsearch/11.0.1
+    "Logstash/#{LOGSTASH_VERSION} (OS=#{os_name}-#{os_version}-#{os_arch}; JVM=#{jvm_vendor}-#{jvm_version}) logstash-#{@plugin_type}-#{config_name}/#{plugin_version}"
```

Comment:
Do we have examples of how this will be displayed on Windows/BSD/etc? And is the example for OS= an accurate reflection?

What I see in the OS section is basically the kernel version, but not a proper OS name/version/arch combo. For example, I would have expected something like:

OS=Ubuntu-18.04-amd64

Reply:
On windows it should be something like `OS=Windows 8.1-6.3-amd64`.
On Mac OS `OS=Mac OS X-10.7.4-amd64`
But on BSD I don't know

Reply:
@andsel one more nitpick here. The 'Logstash' at the start of the line should likely be lowercase.